src/dawn/IIR/Stencil.cpp

```diff
@@ -100,20 +116,21 @@ Stencil::Stencil(StencilInstantiation& stencilInstantiation,
 
 std::unordered_set<Interval> Stencil::getIntervals() const {
   std::unordered_set<Interval> intervals;
-  for(const auto& multistage : multistages_)
-    for(const auto& stage : multistage->getStages())
-      for(const auto& doMethod : stage->getDoMethods())
+  for(const auto& multistage : children_)
+    for(const auto& stage : multistage->getChildren())
+      for(const auto& doMethod : stage->getChildren())
         intervals.insert(doMethod->getInterval());
 
   return intervals;
 }
 
 std::vector<Stencil::FieldInfo> Stencil::getFields(bool withTemporaries) const {
   std::set<int> fieldAccessIDs;
-  for(const auto& multistage : multistages_) {
-    for(const auto& stage : multistage->getStages()) {
-      for(const auto& field : stage->getFields()) {
-        fieldAccessIDs.insert(field.getAccessID());
+  for(const auto& multistage : children_) {
```

Comment:
one iterator?

Reply:
what do you mean?

Reply:
is removed with a later pr, just have an iterator over multiple levels of the tree